sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/view/ViewsImpl.scala

```diff
@@ -102,6 +113,8 @@ final class ViewsImpl(system: ActorSystem, _services: Map[String, ViewService])
                   case e: ViewException => throw e
                   case NonFatal(error) =>
                     throw ViewException(context, s"View unexpected failure: ${error.getMessage}", Some(error))
+                } finally {
+                  MDC.remove(Telemetry.TRACE_ID)
```

Comment:
This means we always touch a thread local, even if there was no trace to start with. Maybe keep a boolean around from setting it or not to know if it needs clearing out or not?

Reply:
done